ingestion/src/metadata/mixins/pandas/pandas_mixin.py

```diff
@@ -76,3 +79,56 @@ def get_partitioned_df(self, dfs):
             ]
             for df in dfs
         ]
+
+    def return_ometa_dataframes_sampled(  # pylint: disable=inconsistent-return-statements
+        self, service_connection_config, client, table, profile_sample_config
+    ):
+        """
+        returns sampled ometa dataframes
+        """
+
+        data = ometa_to_dataframe(
+            config_source=service_connection_config.configSource,
+            client=client,
+            table=table,
+        )
+
+        if data:
+            random.shuffle(data)
+
+            # sampling data based on profiler config (if any)
+            if hasattr(profile_sample_config, "profile_sample"):
+                if (
+                    profile_sample_config.profile_sample_type
+                    == ProfileSampleType.PERCENTAGE
+                ):
+                    return [
+                        df.sample(
+                            frac=profile_sample_config.profile_sample / 100,
+                            random_state=random.randint(0, 100),
+                            replace=True,
+                        )
+                        for df in data
+                    ]
+                if profile_sample_config.profile_sample_type == ProfileSampleType.ROWS:
+                    sample_rows_per_chunk: int = math.floor(
+                        profile_sample_config.profile_sample / len(data)
+                    )
+                    return [
+                        df.sample(
+                            n=sample_rows_per_chunk,
+                            random_state=random.randint(0, 100),
+                            replace=True,
+                        )
+                        for df in data
+                    ]
+            else:
+                # randomize the samples
+                return [
+                    df.sample(
+                        frac=1,
+                        random_state=random.randint(0, 100),
+                    )
+                    for df in data
+                ]
+            return data
```

Comment:
Shouldn't we just return the data here if there no sampling config?

Reply:
not sure if i remember correctly, but randomizing the sample was something that we discussed, if i am not wrong.